sql/src/test/java/org/apache/druid/sql/calcite/schema/DruidSchemaTest.java

```diff
@@ -343,7 +357,58 @@ public void testNullAvailableSegmentMetadata() throws IOException
 
     // The following line can cause NPE without segmentMetadata null check in DruidSchema#refreshSegmentsForDataSource
     schema.refreshSegments(segments.stream().map(DataSegment::getId).collect(Collectors.toSet()));
-    Assert.assertEquals(schema.getSegmentMetadataSnapshot().size(), 2);
+    Assert.assertEquals(schema.getSegmentMetadataSnapshot().size(), 3);
+  }
+
+  @Test
+  public void testAvailableSegmentMetadataIsRealtime()
+  {
+    Map<SegmentId, AvailableSegmentMetadata> segmentsMetadata = schema.getSegmentMetadataSnapshot();
+    final List<DataSegment> segments = segmentsMetadata.values()
+                                                       .stream()
+                                                       .map(AvailableSegmentMetadata::getSegment)
+                                                       .collect(Collectors.toList());
+    // find the only realtime segment with datasource "foo3"
+    final DataSegment existingSegment = segments.stream()
+                                                .filter(segment -> segment.getDataSource().equals("foo3"))
+                                                .findFirst()
+                                                .orElse(null);
+
```

Comment:
Do you want to assert `existingSegment` is not `null`? (similar comment for other occurrences below)

Reply:
yup